### PR TITLE
Upgrade ArgoCD to 2.0.4

### PR DESCRIPTION
**Pull request by bastjan**
This PR updates the bootstrapped ArgoCD to `v2.0.4`.

See https://github.com/projectsyn/component-argocd/issues/18.

I switched from `statik` to native go embedding int he same PR. No more forgetting `go generate ./...`.

Tested on a local cluster with

```yaml
parameters:
  components:
    argocd:
      version: upgrade-argocd-to-v2.x
```

## Checklist

- [x] Keep pull requests small so they can be easily reviewed.
- [ ] Update the documentation.
- [x] Categorize the PR by setting a good title and adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -19,6 +17,11 @@ import (
 // Version is the steward version (set during build)
 var Version = "unreleased"
 
+const (
+	DefaultArgoCDImage      = "quay.io/argoproj/argocd:v2.0.4@sha256:976dfbfadb817ba59f4f641597a13df7b967cd5a1059c966fa843869c9463348"
+	DefaultArgoCDRedisImage = "docker.io/redis:6.2.4@sha256:f631ff6c898339306ffdb8369add5c12303ec3946610ef8d6f1d05f575942f0c"
```
Comment (ccremer) [inline main.go:22]: While you're at it, could you try using the Bitnami image on quay.io?
It's an open issue we have anyway (https://ticket.vshn.net/browse/SYN-288)

File: pkg/agent/agent.go [hunk 2]
```diff
@@ -20,14 +20,15 @@ import (
...
-	ArgoCDImage  string
+	APIURL           *url.URL
+	Token            string
+	ClusterID        string
+	CloudType        string
+	CloudRegion      string
+	Distribution     string
+	Namespace        string
+	ArgoCDImage      string
+	ArgoCDRedisImage string
```
Comment (ccremer) [inline pkg/agent/agent.go:31]: I understand the reason why you'd go for that name, but it's also unlikely that we add more different Redises anyway
```suggestion
	RedisImage string
```
(suggestion also applies to the rest of the code, like flags, etc.)

File: pkg/argocd/argocd.go [hunk 3]
```diff
@@ -62,10 +61,10 @@ func Apply(ctx context.Context, config *rest.Config, namespace, argoImage string
 	}
 
 	klog.Infof("Found %d of expected %d deployments, found %d of expected %d statefulsets, bootstrapping now", foundDeploymentCount, expectedDeploymentCount, foundStatefulSetCount, expectedStatefulSetCount)
-	return bootstrapArgo(ctx, clientset, config, namespace, argoImage, apiClient, cluster)
+	return bootstrapArgo(ctx, clientset, config, namespace, argoImage, redisArgoImage, apiClient, cluster)
```
[the remark this answers is not in the document]
Comment (bastjan) [reply pkg/argocd/argocd.go:64]: yep, improved :)

**Review by ccremer — APPROVED** [review]

File: main.go (hunk 1 above)
Comment (bastjan) [reply main.go]: Tracked in #80 .

**Pull request merged**